src/sentry/search/snuba/executors.py

```diff
@@ -1838,6 +1850,15 @@ def query(
                 count += bulk_result[k]["data"][0]["count"]
             k += 1
 
+        if too_many_candidates and group_ids_to_pass_to_snuba == []:
```

Comment:
```suggestion
        if too_many_candidates:
```

No need to check group_ids_to_pass_to_snuba, when too_many_candidates it's not relevant.